extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/MCPProcessor.java

```diff
@@ -0,0 +1,86 @@
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.devui.deployment.InternalPageBuildItem;
+import io.quarkus.devui.runtime.DevUIRecorder;
+import io.quarkus.devui.runtime.mcp.MCPResourcesService;
+import io.quarkus.devui.runtime.mcp.MCPToolsService;
+import io.quarkus.devui.spi.JsonRPCProvidersBuildItem;
+import io.quarkus.devui.spi.page.Page;
+import io.quarkus.vertx.core.deployment.CoreVertxBuildItem;
+import io.quarkus.vertx.http.deployment.NonApplicationRootPathBuildItem;
+import io.quarkus.vertx.http.deployment.RequireBodyHandlerBuildItem;
+import io.quarkus.vertx.http.deployment.RouteBuildItem;
+
+public class MCPProcessor {
+
+    private static final String DEVMCP = "dev-mcp";
+
+    private static final String NS_MCP = "mcp";
+    private static final String NS_RESOURCES = "resources";
+    private static final String NS_TOOLS = "tools";
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    InternalPageBuildItem createMCPPage(NonApplicationRootPathBuildItem nonApplicationRootPathBuildItem) {
+        InternalPageBuildItem mcpServerPage = new InternalPageBuildItem("MCP Server", 80);
+
+        // Pages
+        mcpServerPage.addPage(Page.webComponentPageBuilder()
+                .namespace(NS_MCP)
+                .title("MCP Server")
+                .icon("font-awesome-solid:robot")
+                .componentLink("qwc-mcp-server.js"));
+
+        mcpServerPage.addPage(Page.webComponentPageBuilder()
+                .namespace(NS_MCP)
+                .title("Tools")
+                .icon("font-awesome-solid:screwdriver-wrench")
+                .componentLink("qwc-mcp-tools.js"));
+
+        mcpServerPage.addPage(Page.webComponentPageBuilder()
+                .namespace(NS_MCP)
+                .title("Resources")
+                .icon("font-awesome-solid:file-invoice")
+                .componentLink("qwc-mcp-resources.js"));
+
+        return mcpServerPage;
+    }
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    @io.quarkus.deployment.annotations.Record(ExecutionTime.RUNTIME_INIT)
+    void registerDevUiHandlers(
+            CoreVertxBuildItem vertx,
+            BuildProducer<RouteBuildItem> routeProducer,
+            DevUIRecorder recorder,
+            LaunchModeBuildItem launchModeBuildItem,
+            BuildProducer<RequireBodyHandlerBuildItem> requireBodyHandlerProducer, // TODO: is this really needed to read the request body?
+            NonApplicationRootPathBuildItem nonApplicationRootPathBuildItem) throws IOException {
+
+        if (launchModeBuildItem.isNotLocalDevModeType()) {
+            return;
+        }
+
+        // SSE for JsonRPC comms
+        routeProducer.produce(
+                nonApplicationRootPathBuildItem
+                        .routeBuilder().route(DEVMCP)
+                        .handler(recorder.streamableHttpHandler(vertx.getVertx()))
+                        .build());
+        requireBodyHandlerProducer.produce(new RequireBodyHandlerBuildItem());
```

Comment:
@phillip-kruger @cescoffier is this really needed for my route to be able to read the HTTP body?

Reply:
Yes, as it ensures we only have one body handler.